core/thread/auth.go

```diff
@@ -0,0 +1,155 @@
+package thread
+
+import (
+	"context"
+	"encoding"
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"github.com/gogo/status"
+	"github.com/grpc-ecosystem/go-grpc-middleware/util/metautils"
+	"github.com/libp2p/go-libp2p-core/crypto"
+	"google.golang.org/grpc/codes"
+)
+
+type Identity interface {
```

Comment:
@carsonfarmer This provides the ability to use a generic "identity" that can sign. Most of the time this would just be a private key, which implements this interface.